docs/source/subgraphs.md

```diff
@@ -52,14 +52,23 @@ a `.graphql` SDL file, you can pass the path to that file as the `--schema` flag
 (or `-s` for short):
 
 ```bash
-rover subgraph push my-graph --schema ./accounts/schema.graphql --service-name accounts
+rover subgraph push my-graph\
+  --schema ./accounts/schema.graphql\
+  --service-name accounts\
+  --routing-url https://my-running-service.com/api
 ```
 
 this will push the schema at ./accounts/schema.graphql to the default (`current`) variant
 of `my-graph`. You can specify a different variant by adding the variant name
 after the graph name like `my-graph@variant-name`. The `--service-name` flag
 just tells the registry which subgraph you're pushing changes to.
 
+The `--routing-url` is used by a gateway running in [managed federation mode]
+(https://www.apollographql.com/docs/federation/managed-federation/overview/). If
+You're running a service which hasn't been deployed yet or isn't using managed
+federation, you may pass a placeholder url or leave the flag empty
+(i.e. `--routing-url=""`).
```

Comment:
This seems like something that would benefit from some guidance in Rover itself.

I propose that if they pass a `--routing-url` to a `subgraph push` invocation that doesn't need it (in the case of a non-deployed service, or unmanaged federation), Rover should return an error that lets them know that what they passed won't be used. I think this approach would eliminate some of the confusion our users may have about the usage of the flag.

Reply:
I actually think I missed Josh's original comment here (or just forgot since it's been a while), but it seems valuable here:

> If they are pushing partial schemas to the registry then it will be managed federation. Unmanaged federation doesn't go through the registry.
Unmanaged federation directly get the subgraph schemas by querying the urls at runtime and fetching them via introspection and composing it directly in the gateway. The registry is bypassed in this mode.

---

If this is the case, then perhaps we don't need to mention unmanaged federation in this section of the docs? Perhaps my confusion is coming from the suggestion of the `--routing-url=""` here

Reply:
@EverlastingBugstopper There's no way for the registry or rover to know if a gateway is running in managed federation mode though unfortunately.

Managed federation is chosen at runtime on the gateway. it's just a matter of where the gateway is getting its config from: either a list of service urls or from studio. So there's no way for us to figure this out here ☹️

Reply:
We could make this flag optional, but it's not optional in the API. So we'd have to pass the empty string through to studio anyway